### PR TITLE
Add support for Shyft Network (SHFT)

### PR DESCRIPTION
**Pull request by noeleo**
homepage: https://shyft.network
block explorer: https://bx.shyft.network/
slip0044 index: 7341
chain ID: 7341

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -211,9 +211,12 @@ else ifeq ($(CHAIN),songbird)
 APP_LOAD_PARAMS += --path "44'/554'" --path "44'/60'"
 DEFINES += CHAINID_UPCASE=\"SONGBIRD\" CHAINID_COINNAME=\"SGB\" CHAIN_KIND=CHAIN_KIND_SONGBIRD CHAIN_ID=19
 APPNAME = "Songbird"
-else
+else ifeq ($(CHAIN),shyft)
+APP_LOAD_PARAMS += --path "44'/7341'"
+DEFINES += CHAINID_UPCASE=\"SHYFT\" CHAINID_COINNAME=\"SHFT\" CHAIN_KIND=CHAIN_KIND_SHYFT CHAIN_ID=7341
+APPNAME = "Shyft"
```
Comment (diwu1989) [inline Makefile:217]: missing `else` at the end

**Comment by diwu1989** [comment]
Video showing this working with Metamask:
1. shows verifying public address on ledger
2. shows transaction information on ledger
3. shows rejecting a transaction on ledger
4. shows accepting a transaction on ledger, and shows successful transaction on SHFT block explorer
https://user-images.githubusercontent.com/380950/136710279-34bab4d0-5b99-45e8-8d6a-7e4725826ad9.mp4

File: Makefile (hunk 1 above)
Comment (pscott) [reply Makefile:236]: can I ask why you removed the `else` at the end? :)

**Review by pscott — CHANGES_REQUESTED** [review]
Missing `else` in the Makefile :)

File: Makefile (hunk 1 above)
Comment (noeleo) [reply Makefile:236]: mistake! done

**Review by pscott — APPROVED** [review]

**Comment by diwu1989** [comment]
I was able to rebuild the SHYFT app and re-install it on my ledger.

There is something weird where if I open the SHYFT app, metamask doesn't seem to connect and won't work with it.
But if I open the Ethereum app, it works fine with Metamask and can send and sign transactions for Shyft.

Here's another test transaction:
https://bx.shyft.network/tx/0x2d9a09861e2e6a88677a5c0b63dcb20686d64d8e6230caa74f86832bc783515f/internal-transactions
image
image
image
image

**Comment by diwu1989** [comment]
I think here's whats going on:
1. metamask has a hardcoded derivation path (the one for ethereum)
2. the rebuilt base Ethereum app has the ability to sign for Shyft network ID correctly, so that's why I can sign Shyft transactions with Eth app
3. Shyft app can't sign transactions from Metamask cus of mismatched derivation path (and metamask doesn't seem to let you override derivation path)

At least all of this testing proves that the underlying Shyft network support code works fine, it's just that there's a ghetto derivation path mismatch that prevents Metamask from using the Shyft app directly.

**Comment by noeleo** [comment]
Di, should we just add the ethereum derivation path to the code? I think
it’s 60

**Comment by diwu1989** [comment]
@noeleo @pscott 
I can confirm that once the derivation path is changed to match ETH, Metamask is now able to use the Shyft app to sign and send transaction.

The only change we need is to the makefile:
```
diff --git a/Makefile b/Makefile
index 6d597db..ab19944 100644
--- a/Makefile
+++ b/Makefile
@@ -231,7 +231,7 @@ APP_LOAD_PARAMS += --path "44'/554'" --path "44'/60'"
 DEFINES += CHAINID_UPCASE=\"SONGBIRD\" CHAINID_COINNAME=\"SGB\" CHAIN_KIND=CHAIN_KIND_SONGBIRD CHAIN_ID=19
 APPNAME = "Songbird"
 else ifeq ($(CHAIN),shyft)
-APP_LOAD_PARAMS += --path "44'/7341'"
+APP_LOAD_PARAMS += --path "44'/60'"
 DEFINES += CHAINID_UPCASE=\"SHYFT\" CHAINID_COINNAME=\"SHFT\" CHAIN_KIND=CHAIN_KIND_SHYFT CHAIN_ID=7341
 APPNAME = "Shyft"
 else ifeq ($(CHAIN),polygon)
```

This is a test transaction I just sent with the newly rebuilt app:
https://bx.shyft.network/tx/0x2956b70e324fcc086a3da61db9830d838a39eccee1d911566120bc1d5fd0df5d/internal-transactions

Inside the app, when I approve the transaction, the currency shown is still ETH, but it does go through as SHFYT

**Comment by diwu1989** [comment]
Video showing a transaction end to end
https://bx.shyft.network/tx/0x42a8b5f759291eab3d838b3f26eab09a25297bd91aca8fc20eacf4db7fdf931c/internal-transactions

https://user-images.githubusercontent.com/380950/144805882-e5216a2d-8142-456b-af78-34c2e352be7d.mp4

**Comment by FabriceDautriat** [comment]
Since your chainID is different from 0 or 1, then it's ok to add 44'/60' derivation path.

https://developers.ledger.com/docs/nano-app/eth-clones/

Optionally you can also add your network detail in https://github.com/LedgerHQ/app-ethereum/blob/develop/src_common/network.c

**Comment by tjulien-ledger** [comment]
Hi @diwu1989 you can indeed add the path following in this manner while keeping your own path:
`APP_LOAD_PARAMS += --path "44'/7341'" --path "44'/60'"`
Regarding the currency being displayed in ETH instead of SHFT, I think you may not have run `make clean` after building the ETH app, I have tested your code adding the ETH path on my side and the currency is displayed as SHFT not ETH. 
Lastly, there is more recent PR open, can you make the change there instead please?
https://github.com/LedgerHQ/app-ethereum/pull/210

**Pull request closed without merging**